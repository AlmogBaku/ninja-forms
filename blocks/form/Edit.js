import React from "react";
import {
	Placeholder,
	ServerSideRender,
	SelectControl
} from "@wordpress/components";
import { InspectorControls } from "@wordpress/block-editor";
import useForms from "./useForms";

/**
 * Form chooser select control
 */
export const ChooseForm = ({ formId, forms, onChange, labelText }) => {
	const { asSelectOptions } = useForms({ forms });
	const options = React.useMemo(() => {
		let opts = asSelectOptions();
		return [...[{ label: "---" }], ...opts];
	}, [forms, asSelectOptions]);
	return (
		<SelectControl
			label={labelText ? labelText : "Choose Form"}
			value={formId}
			onChange={onChange}
			options={options}
		/>
	);
};

/**
 * Ninja Forms block UI
 */
export default function Edit({
	formId,
	formTitle,
	forms,
	setAttributes,
	labelText
}) {
	const { getFormTitle } = useForms({ forms });
	const updateChosenForm = formId => {
		setAttributes({
			formId: formId,
			formTitle: getFormTitle(formId)
		});
	};

	if (!formId) {
		return (
			<Placeholder>
				<ChooseForm
					label={labelText ? labelText : "Choose Form"}
					formId={formId}
					forms={forms}
					onChange={updateChosenForm}
					labelText={labelText}
				/>
			</Placeholder>
		);
	}

	return (
		<React.Fragment>
			<InspectorControls>
				<ChooseForm
					formId={formId}
					forms={forms}
					onChange={updateChosenForm}
					labelText={labelText}
				/>
			</InspectorControls>
			<ServerSideRender
				block="ninja-forms/form"
				attributes={{
					formId,
					formTitle
				}}
			/>
		</React.Fragment>
	);
}
