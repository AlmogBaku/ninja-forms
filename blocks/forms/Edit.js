import React from 'react';

export default  function Edit() {
    return <React.Fragment/>
}