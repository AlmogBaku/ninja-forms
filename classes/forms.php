<?php
/**
 * Handles adding and removing forms.
 *
 * @package     Ninja Forms
 * @subpackage  Classes/Form
 * @copyright   Copyright (c) 2014, WPNINJAS
 * @license     http://opensource.org/licenses/gpl-2.0.php GNU Public License
 * @since       2.9
*/

class NF_Forms {

	/**
	 * Get things started
	 * 
	 * @access public
	 * @since 2.9
	 * @return void
	 */
	public function __construct() {
		
	}

	/**
	 * Get all forms
	 * 
	 * @access public
	 * @since 2.9
	 * @return array $forms
	 */
	public function get_all( $debug = false, $show_new = false ) {
		$forms = nf_get_objects_by_type( 'form' );

		$tmp_array = array();
		foreach ( $forms as $form ) {
			$form_id = $form['id'];

			$status = Ninja_Forms()->form( $form_id )->get_setting( 'status' );
			if ( ( $status == 'new' && $show_new ) || $status != 'new' ) {
				$title = Ninja_Forms()->form( $form_id )->get_setting( 'form_title' );
				if ( strpos( $title, '_' ) === 0 ) {
					if ( $debug )
						$tmp_array[] = $form_id;
				} else {
					$tmp_array[] = $form_id;
				}
			}
		}
		return $tmp_array;
	}
}