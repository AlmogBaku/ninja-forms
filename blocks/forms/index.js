import Edit from './Edit';
import  block from './block';
import { __ } from '@wordpress/i18n';
import { registerBlockType } from '@wordpress/blocks';

registerBlockType('ninja-forms/form',{
    ...block,
    title: __('Ninja Form', 'ninja-forms'),
    edit: ({attributes,setAttributes}) => {

        return <Edit/>
    },
    save: () => null
});