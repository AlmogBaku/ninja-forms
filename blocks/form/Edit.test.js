import Edit, { ChooseForm } from "./Edit";
import { render, fireEvent } from "@testing-library/react";
import forms from "./forms.fixture.js";

describe("Form block Edit callback", () => {
	it.skip("Matches snapshot", () => {
		const onChange = jest.fn();

		const { container } = render(
			<Edit
				onChange={onChange}
				formId={forms["1"].formId}
				formTitle={forms["1"].formTitle}
				forms={forms}
			/>
		);
		expect(container).toMatchSnapshot();
	});

	it.skip("Calls onChange when changing form", () => {
		const onChange = jest.fn();
		const { getByLabelText } = render(
			<Edit
				onChange={onChange}
				formId={forms[1].formId}
				formTitle={forms[1].formTitle}
				forms={forms}
			/>
		);
		fireEvent.change(getByLabelText("Form"), {
			target: { value: "3" }
		});
		expect(onChange).toBeCalledTimes(1);
		expect(onChange).toBeCalledWith("3");
	});

	it.skip("Shows place holder when no formId passed", () => {
		const onChange = jest.fn();
		const { container } = render(<Edit onChange={onChange} forms={forms} />);
		expect(container).toBe("I do not know?");
	});

	it.skip("Shows serve-side-render when formId is passed", () => {
		const onChange = jest.fn();
		const { container } = render(
			<Edit
				onChange={onChange}
				formId={forms[0].formId}
				formTitle={forms[1].formTitle}
				forms={forms}
			/>
		);
		expect(container).toBe("I do not know?");
	});
});

describe("ChooseForm", () => {
	it.skip("matches snapshot", () => {
		const { container } = render(
			<ChooseForm formId={2} forms={forms} onChange={jest.fn()} />
		);
		expect(container).toMatchSnapshot();
	});
	it.skip("Calls on change with chosen form id", () => {
		const onChange = jest.fn();
		const labelText = "Test";
		const { getByLabelText } = render(
			<ChooseForm
				formId={2}
				forms={forms}
				onChange={onChange}
				labelText={labelText}
			/>
		);
		fireEvent.change(getByLabelText(labelText), {
			target: { value: "3" }
		});
		expect(onChange).toBeCalledTimes(1);
		expect(onChange).toBeCalledWith("3");
	});
});
