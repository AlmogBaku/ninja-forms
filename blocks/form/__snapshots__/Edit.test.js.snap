// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form block Edit callback Renders form preview 1`] = `<div />`;

exports[`Form block Edit callback Renders placeholder 1`] = `
<div>
  <div
    class="components-placeholder is-small"
  >
    <iframe
      aria-hidden="true"
      aria-label="resize-listener"
      frameborder="0"
      src="about:blank"
      style="display: block; opacity: 0; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;"
      tabindex="-1"
    />
    <div
      class="components-placeholder__label"
    />
    <div
      class="components-placeholder__fieldset"
    >
      <div
        class="components-base-control"
      >
        <div
          class="components-base-control__field"
        >
          <label
            class="components-base-control__label"
            for="inspector-select-control-0"
          >
            Choose Form
          </label>
          <select
            class="components-select-control__input"
            id="inspector-select-control-0"
          >
            <option>
              ---
            </option>
            <option
              value="1"
            >
              Fake Form
            </option>
            <option
              value="2"
            >
              Real Form
            </option>
            <option
              value="3"
            >
              Unreal Form
            </option>
            <option
              value="4"
            >
              Mystery Form
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</div>
`;
